presentation/src/main/java/com/architect/coders/mu8/comics/ComicsActivity.kt

```diff
@@ -51,35 +42,45 @@ class ComicsActivity : AppCompatActivity() {
             )
         }
 
+        binding.viewmodel = viewModel
+        binding.lifecycleOwner = this
+
+        configureToolbar()
+
+        configureRecycler()
+
+        observerViewModel()
+    }
+
+    private fun configureToolbar() {
+        toolbar = findViewById(R.id.comic_activity_toolbar)
+        setSupportActionBar(toolbar)
+        supportActionBar?.setDisplayShowTitleEnabled(false)
+
+        toolbarTitle = findViewById(R.id.toolbar_title)
+        toolbarTitle.text = getString(R.string.comics_name)
+    }
+
+    private fun configureRecycler() {
         adapter = ComicsAdapter(viewModel::onComicClicked)
         recycler.layoutManager = GridLayoutManager(this, GRID_COLUMS)
         recycler.adapter = adapter
+    }
 
-        viewModel.model.observe(this, Observer(::updateUi))
+    private fun observerViewModel() {
+        viewModel.comics.observe(this, Observer(::showComics))
+        viewModel.messageError.observe(this, Observer(::showMessages))
+        viewModel.navegateTo.observe(this, EventObserver { comic ->
+            toast(comic.title)
+        })
     }
 
-    private fun updateUi(uiModel: ComicsViewModel.UiModel) {
-        return when (uiModel) {
-            is ShowLoading -> {
-                progress.makeItVisible()
-                recycler.makeItGone()
-            }
-            is LoadData -> {
-                adapter.comics = uiModel.comics
-                progress.makeItGone()
-                recycler.makeItVisible()
-            }
-            is InternetError -> {
-                progress.makeItGone()
-                recycler.makeItGone()
-                Snackbar.make(
-                    recycler,
-                    getString(R.string.error_internet_message),
-                    Snackbar.LENGTH_LONG
-                ).show()
-            }
-            is NavigateTo -> Toast.makeText(this, uiModel.comic.title, Toast.LENGTH_LONG).show()
-        }
+    private fun showComics(comics: List<Comic>) {
+        adapter.comics = comics
+    }
+
+    private fun showMessages(message: String) {
```

Comment:
Rename as show message in singular, you are just receiving one parameter not multiple ones

Reply:
Done